app/src/main/java/com/bc/gordiansigner/service/AccountMapService.kt

```diff
@@ -0,0 +1,49 @@
+package com.bc.gordiansigner.service
+
+import com.bc.gordiansigner.helper.ext.newGsonInstance
+import com.bc.gordiansigner.model.AccountMap
+import com.bc.gordiansigner.model.Descriptor
+import com.bc.gordiansigner.model.HDKey
+import com.bc.gordiansigner.service.storage.FileStorageApi
+import com.bc.gordiansigner.service.storage.SharedPrefApi
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+class AccountMapService @Inject constructor(
+    sharedPrefApi: SharedPrefApi,
+    fileStorageApi: FileStorageApi
+) : BaseService(sharedPrefApi, fileStorageApi) {
+
+    fun saveAccountMap(accountMap: AccountMap): Single<AccountMap> {
+        //TODO: save account map, implement later
+
+        return Single.just(accountMap)
+    }
+
+    fun getAccountMapInfo(accountMapString: String): Single<Pair<AccountMap, Descriptor>> =
+        Single.fromCallable {
+            val accountMap = newGsonInstance().fromJson(accountMapString, AccountMap::class.java)
+            val descriptor = Descriptor.fromString(accountMap.descriptor)
+
+            Pair(accountMap, descriptor)
+        }.subscribeOn(Schedulers.io())
```

Comment:
It should not be scheduled on IO scheduler since it does not communicate with any IO stuff. You can consider `computation` or even `newThread` if needed.